From 66335b4d15625e8963cba6b3214da89da325014e Mon Sep 17 00:00:00 2001
From: Andrew Branson <andrew.branson@jollamobile.com>
Date: Wed, 15 Nov 2017 12:25:37 +0100
Subject: [PATCH 12/15] Skip OMX header version if not defined

---
 omx/gstomx.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/omx/gstomx.h b/omx/gstomx.h
index 7d3b892..3941c9c 100644
--- a/omx/gstomx.h
+++ b/omx/gstomx.h
@@ -68,6 +68,7 @@
 
 G_BEGIN_DECLS
 
+#ifdef OMX_VERSION_MAJOR
 #define GST_OMX_INIT_STRUCT(st) G_STMT_START { \
   memset ((st), 0, sizeof (*(st))); \
   (st)->nSize = sizeof (*(st)); \
@@ -76,6 +77,12 @@ G_BEGIN_DECLS
   (st)->nVersion.s.nRevision = OMX_VERSION_REVISION; \
   (st)->nVersion.s.nStep = OMX_VERSION_STEP; \
 } G_STMT_END
+#else
+#define GST_OMX_INIT_STRUCT(st) G_STMT_START { \
+  memset ((st), 0, sizeof (*(st))); \
+  (st)->nSize = sizeof (*(st)); \
+} G_STMT_END
+#endif
 
 #ifdef OMX_SKIP64BIT
 #define GST_OMX_GET_TICKS(ticks) ((((guint64) (ticks).nHighPart) << 32) | ((ticks).nLowPart))
-- 
1.8.3-rc3

