From 17a41b48141b3eaf44fb8aa7dfa78f323ff637cc Wed, 15 Nov 2017 12:27:33 +0100
From: Andrew Branson <andrew.branson@jollamobile.com>
Date: Wed, 15 Nov 2017 12:25:37 +0100
Subject: [PATCH] Skip OMX header version if not defined

diff --git a/omx/gstomx.h b/omx/gstomx.h
index 60a315b..f18636f 100644
--- a/omx/gstomx.h
+++ b/omx/gstomx.h
@@ -68,6 +68,7 @@
 
 G_BEGIN_DECLS
 
+#ifdef OMX_VERSION_MAJOR
 #define GST_OMX_INIT_STRUCT(st) G_STMT_START { \
   memset ((st), 0, sizeof (*(st))); \
   (st)->nSize = sizeof (*(st)); \
@@ -76,6 +77,12 @@
   (st)->nVersion.s.nRevision = OMX_VERSION_REVISION; \
   (st)->nVersion.s.nStep = OMX_VERSION_STEP; \
 } G_STMT_END
+#else
+#define GST_OMX_INIT_STRUCT(st) G_STMT_START { \
+  memset ((st), 0, sizeof (*(st))); \
+  (st)->nSize = sizeof (*(st)); \
+} G_STMT_END
+#endif
 
 #ifdef OMX_SKIP64BIT
 #define GST_OMX_GET_TICKS(ticks) ((((guint64) (ticks).nHighPart) << 32) | ((ticks).nLowPart))
